IGraph.h

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright 2004-2015 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ * 
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * 
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+//===----------------------------------------------------------------------===//
+// Chapel LLVM Locality Optimization by Akihiro Hayashi (ahayashi@rice.edu)
+//===----------------------------------------------------------------------===//
+// SSA Value Graph for Locality Inference
+//===----------------------------------------------------------------------===//
```

Comment:
Is this called IGraph for Inequality Graph? Could you say more about the general algorithm/data structure in a comment here?